pkg/spiffe/spiffe.go

```diff
@@ -99,6 +101,34 @@ func MustGenSpiffeURI(ns, serviceAccount string) string {
 	return uri
 }
 
+// ExpandWithTrustDomains expands a given spiffe identities, plus a list of truts domain aliases.
+// We ensure the returned list does not contain duplicates; the original input is always retained.
+// For example,
+// ExpandWithTrustDomains({"spiffe://td1/ns/def/sa/def"}, {"td1", "td2"}) returns
+//   {"spiffe://td1/ns/def/sa/def", "spiffe://td2/ns/def/sa/def"}.
+// ExpandWithTrustDomains({"spiffe://td1/ns/def/sa/a", "spiffe://td1/ns/def/sa/b"}, {"td2"}) returns
+//   {"spiffe://td1/ns/def/sa/a", "spiffe://td2/ns/def/sa/a", "spiffe://td1/ns/def/sa/b", "spiffe://td2/ns/def/sa/b"}.
+func ExpandWithTrustDomains(spiffeIdentities, trustDomainAliases []string) map[string]struct{} {
+	out := map[string]struct{}{}
+	for _, id := range spiffeIdentities {
+		out[id] = struct{}{}
+		// Expand with aliases set.
+		m := spiffePattern.FindStringSubmatchIndex(id)
+		// FindStringSubmatchIndex the pairs of the match, (trust domain + the whole match string) x 2
+		// so we should see 4 index.
+		if len(m) < 4 {
+			spiffeLog.Errorf("Failed to extract SPIFFE trust domain from: %v, match %v", id, m)
+			continue
+		}
+		suffix := id[m[3]:]
+		for _, td := range trustDomainAliases {
+			nid := fmt.Sprintf("%v://%v%v", Scheme, td, suffix)
```

Comment:
avoid sprintf if possible. See comment above, probably makes sense to have a function that bi-directionally converts spiffe string to spiffe structure and use that here

Reply:
using `+` as  instead? https://dev.to/pmalhaire/concatenate-strings-in-golang-a-quick-benchmark-4ahh

and like the rest of the code?

Reply:
Yeah - this might be a premature optimization though, not 100% sure tbh. I think this code is on the hot path but it may not be